proxy.go

```diff
@@ -121,13 +121,46 @@ func setupLogger(logLevel string) error {
 	return nil
 }
 
+func printAgentLogs(sock string) error {
+	// Don't return an error if nothing has been provided.
+	// This flag is optional.
+	if sock == "" {
+		return nil
+	}
+
+	agentLogsAddr, err := unixAddr(sock)
+	if err != nil {
+		proxyLog.Fatal("invalid agent logs socket address")
+		return err
+	}
+
+	conn, err := net.Dial("unix", agentLogsAddr)
```

Comment:
Well yeah, we have no way to know when to close it since the proxy is going to be killed. The socket will be closed at this time. I agree this is not the prettiest way to handle this. We could put a signal handler in the proxy, assuming this is going to be terminated with a SIGTERM. This way, we could properly cleanup.
This will be done in a separate PR if this is what we want.